parquet-column/src/main/java/org/apache/parquet/io/MessageColumnIO.java

```diff
@@ -276,11 +276,11 @@ private void printState() {
 
     private void log(Object message, Object...parameters) {
       if (DEBUG) {
-        String indent = "";
+        StringBuilder indent = new StringBuilder();
```

Comment:
You may create the builder with an exact capacity so it does not need to grow during appending values.
Also, you may count in the message itself and appending it instead of concatenating the indent to the message separately.